common/src/main/java/com/gregtechceu/gtceu/api/data/chemical/ChemicalHelper.java

```diff
@@ -242,4 +263,25 @@ public static List<Map.Entry<ItemStack, ItemMaterialInfo>> getAllItemInfos() {
                 .map(entry -> new AbstractMap.SimpleEntry<>(new ItemStack(entry.getKey().asItem()), entry.getValue()))
                 .collect(Collectors.toList());
     }
+
+    public static Ingredient getValidItemsForTag(TagKey<Item> tag) {
+        var item = AlmostUnifiedAdapter.getPreferredItemForTag(tag);
+        if (item != null) {
+            return Ingredient.of(item);
+        } else {
+            return Ingredient.of(tag);
+        }
+    }
+
+    private static class AlmostUnifiedAdapter {
+
+        @Nullable
+        public static Item getPreferredItemForTag(TagKey<Item> tag) {
```

Comment:
In case this gets called in tick logic, it might be better (in terms of performance) to make this an interface and add a static variable with the correct implementation.  
Could also just use a `Function<TagKey<Item>, Item>` for that though.

Of course that depends on the performance impact of `GTCEu.isAlmostUnifiedLoaded()`.

If it's only called in our initialization code, I'd keep it as-is.